### ISSUE TITLE
Don't always close window on right click

### ISSUE THREAD

**Issue by tothxa** [widelands/widelands#6242]
**Is your feature request related to a problem? Please describe.**
While the possibility to quickly close windows with right-clicking them is extremely convenient, it sometimes breaks user expectations. E.g.: [cutting and pasting text](https://www.widelands.org/forum/post/41347/)

**Describe the solution you'd like**
Don't close windows when a multiline input box is right clicked? Maybe even add a context menu for cut and paste?

Have a way to disable close-on-right-click for some windows where it makes less sense? Like most main menu windows? Maybe by default-pinning them? Or just add a close-confirmation dialog?

**Implementation steps**
- [ ] Decide where this should be done
- [ ] Decide how it should be done
- [ ] Do it :)

**Comment by bunnybot** [comment]
<img align='left' src='https://codeberg.org/avatars/a4ad2f9a2504f052397d8ba82ed48ba1' alt='Nordfriese' height='48' />*[Mirrored from Codeberg](https://codeberg.org/wl/widelands/issues/4583#issuecomment-1357601)*
*On Tue Nov 28 17:47:48 CET 2023, **Benedikt Straub (Nordfriese)** wrote:*

---

+1 for consuming right-clicks in text input panels. This part is easy.
+1 for a "cut/copy/paste" context menu, although harder.
-1 for default-pinning any windows, a window that makes itself hard to close is an annoyance.

**Comment by bunnybot** [comment]
Assigned to Nordfriese
